### PR TITLE
Handle contextmenu and touchevents correctly

### LINKED ISSUE #3749: Firefox: can't open contextpopup
The touch events are activated for contextopup and tooltip on firefox desktop.

### PR DESCRIPTION
**Pull request by oterral**
Fix #3749 

[test](https://mf-geoadmin3.int.bgdi.ch/teo_fixes/index.html?topic=ech&lang=de&bgLayer=ch.swisstopo.pixelkarte-farbe&layers=ch.swisstopo.zeitreihen,ch.bfs.gebaeude_wohnungs_register,ch.bav.haltestellen-oev,ch.swisstopo.swisstlm3d-wanderwege&layers_visibility=false,false,false,false&layers_timestamp=18641231,,,&X=179307.26&Y=632153.45&zoom=3)

This PR fixes the tooltip and the context popup on hybrid devices (touch and mouse).

### CODE REVIEW

File: src/components/contextpopup/ContextPopupDirective.js [hunk 1]
```diff
@@ -162,32 +180,52 @@ goog.require('ga_what3words_service');
               }
 
               overlay.setPosition(coord21781);
-              showPopover();
+              element.show();
+              // We use a boolean instead of  jquery .is(':visible') selector
+              // because that doesn't work with phantomJS.
+              isPopoverShown = true;
```
Comment (gjn) [inline src/components/contextpopup/ContextPopupDirective.js]: I prefer to initialize such variables that are used in different places (even though they are not needed). It better establishes their scope.

Comment (gjn) [inline src/components/contextpopup/ContextPopupDirective.js]: There's maybe also a way to get rid of it alltogehter and use the state of the element directly.

File: src/components/map/MapService.js [hunk 2]
```diff
@@ -323,65 +327,55 @@ goog.require('ga_urlutils_service');
...
-                  }, 350, false);
-                }
-              }
-              moving = false;
+            // We authorize a move of 5 pixels max for touch device.
+            if (down &&
+                (evt.clientX < down.clientX - 5 ||
+                evt.clientX > down.clientX + 5 ||
+                evt.clientY < down.clientY - 5 ||
+                evt.clientY > down.clientY + 5)) {
```
Comment (gjn) [inline src/components/map/MapService.js]: Interesting approach!

File: src/components/contextpopup/ContextPopupDirective.js (hunk 1 above)
Comment (oterral) [inline src/components/contextpopup/ContextPopupDirective.js]: Yes the easiest way was the jquery stuff but phantomjs doesn't like that.

**Comment by gjn** [comment]
Thank you for the adaptions.

**Pull request merged**